addons/account/models/account_move.py

```diff
@@ -4321,8 +4321,8 @@ def create_tax_cash_basis_entry(self, percentage_before_rec):
                                 'tax_exigible': True,
                                 'tax_ids': [(6, 0, [tax.id])],
                                 'move_id': newly_created_move.id,
-                                'currency_id': line.currency_id.id,
-                                'amount_currency': self.amount_currency and line.currency_id.round(line.amount_currency * amount / line.balance) or 0.0,
+                                'currency_id': self.currency_id.id,
+                                'amount_currency': self.amount_currency and self.currency_id.round(-self.amount_currency * amount / line.balance) or 0.0,
```

Comment:
replacing `line` by `self` seems completely wrong...

Reply:
amount_currency is equal to self.amount_currency and not to line.amount_currency

Reply:
`'amount_currency': line.currency_id.round(line.amount_currency * amount / line.balance) if line.currency_id and line.balance else 0.0,`